src/imitation/util/buffering_wrapper.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+from stable_baselines.common.vec_env import VecEnv, VecEnvWrapper
+
+from imitation.util import rollout
+
+
+class BufferingWrapper(VecEnvWrapper):
+  """Saves transitions of underlying VecEnv.
+
+  Retrieve saved transitions using `pop_transitions()`.
+  """
+
+  def __init__(self, venv: VecEnv, error_on_premature_reset: bool = True):
+    """
+    Args:
+      venv: The wrapped VecEnv.
+      error_premature_reset: Error if `reset()` is called on this wrapper
+        and there are saved samples that haven't yet been accessed.
+    """
+    super().__init__(venv)
+    self.error_on_premature_reset = error_on_premature_reset
+    self._did_reset = False
+    self.obs_list = []
```

Comment:
Weakly prefer these being `None` (unless it causes pytype to start complaining) since must call `reset()` before `step()`